### ISSUE TITLE
Using '--public-access off' when creating storage container causes creation to fail

### ISSUE THREAD

**Issue by usrme** [Azure/azure-cli#13100]
## Describe the bug

**Command Name:**

`az storage container create`

**Errors:**

```
The value for one of the HTTP headers is not in the correct format. ErrorCode: InvalidHeaderValue
<?xml version="1.0" encoding="utf-8"?><Error><Code>InvalidHeaderValue</Code><Message>The value for one of the HTTP headers is not in the correct format.
RequestId:9821f33d-c01e-00e5-78a7-171d7d000000
Time:2020-04-21T06:39:11.6688523Z</Message><HeaderName>x-ms-blob-public-access</HeaderName><HeaderValue>off</HeaderValue></Error>
```

## To Reproduce:

- `az storage container create --name {} --account-name {} --account-key {} --public-access off`

## Expected Behavior

A more meaningful error message stating that setting `public-access` is no longer needed in the latest version.

## Environment Summary

```
Linux-5.5.17-200.fc31.x86_64-x86_64-with-fedora-31-Thirty_One
Python 3.7.6
Installer: RPM

azure-cli 2.4.0
```

Also tested in:

```
Linux-5.0.0-1036-azure-x86_64-with-debian-buster-sid
Python 3.6.5
Installer: DEB

azure-cli 2.4.0
```

## Additional Context

Removing the `public-access` parameter entirely or setting a value of either `blob` or `container` will not cause issues.

**Comment by yonzhan** [comment]
add to S170

**Comment by jsburckhardt** [comment]
I faced this issue today. It was part of script I was running as part of a terraform deployment ... since I couldn't wait till this issue is resolved. I had to find a workaround for a workaround :D. I ended creating the container and then set the permission
'''
az storage container create --name {} --account-name ${}; az storage container set-permission --name {} --account-name ${} --public-access off
'''

**Comment by flytzen** [comment]
@usrme says the expected behaviour is "A more meaningful error message stating that setting public-access is no longer needed in the latest version.". Unfortunately that isn't quite true: Omitting this parameter will give public access by default, which is not the same. 
As @jsburckhardt pointed out, you now have to take two steps. In other words, the fix needs to be to re-instate the support for "off".

**Comment by usrme** [comment]
Oh, damn! I must've missed that as I was sure that the default permission was to deny public access.

**Comment by flytzen** [comment]
@usrme Apologies. I tested it and it looked to me like it didn't set it to "private". However, I have just tried again and I was wrong. Don't know what I was smoking :)

**Comment by jsburckhardt** [comment]
hi guys, I'm not sure it is a feature-request. In the AZ CLI it says one of the options for --public-access is off
![image](https://user-images.githubusercontent.com/18494471/81773807-858e0c00-952c-11ea-9153-28cb5fe7e0ac.png)
When I try to create the resources it complains that the header format is not correct:
![image](https://user-images.githubusercontent.com/18494471/81773890-b1a98d00-952c-11ea-81a5-012cd8d598bd.png)
In the rest api documentation for the container "off" isn't an option
![image](https://user-images.githubusercontent.com/18494471/81774128-38f70080-952d-11ea-94fb-dbe7dba5e37b.png)
I still have to do it in two steps, but I'm not sure for how long I'll be able since security is going to put a policy that containers have to be created with public-access off

**Comment by usrme** [comment]
@jsburckhardt, you don't have to do it in two steps if the default action has changed to not requiring the value `off` to set the public access level to private.

**Comment by Juliehzl** [comment]
dupe of #13373 
Fixed in azure cli 2.6.0
